ts/src/program/namespace/account.ts

```diff
@@ -18,21 +18,22 @@ import Coder, {
 } from "../../coder";
 import { Subscription, Address, translateAddress } from "../common";
 import { getProvider } from "../../";
+import { AllAccountsMap, IdlTypes, TypeDef } from "./types";
 import * as pubkeyUtil from "../../utils/pubkey";
 import * as rpcUtil from "../../utils/rpc";
 
 export default class AccountFactory {
-  public static build(
-    idl: Idl,
+  public static build<IDL extends Idl>(
+    idl: IDL,
     coder: Coder,
     programId: PublicKey,
-    provider: Provider
-  ): AccountNamespace {
+    provider?: Provider
```

Comment:
Is this necessary?

Reply:
There was some recent change that forced this. Some upstream caller doesn't have provider anymore. I think it started at Program level